### PR TITLE
updated editor to Typescript 5.2.2

### PR DESCRIPTION
**Pull request by ggrandi**
- Modified the dl-monaco script to be more automated
- updated ts to 5.2.2

should we also maybe change staging to load monaco from the staging url instead of `typehero.dev`?

### CODE REVIEW

**Comment by Hacksore** [comment]
Obligatory does vim mode work comment hahah

**Comment by bautistaaa** [comment]
added staging to the check. good enough ™

**Review by Hacksore — APPROVED** [review]
staging works™

**Comment by ggrandi** [comment]
Staging is still loading from `https://typehero.dev/min/vs/editor/editor.main.js` but it'll get fixed as soon as we merge (since it'll update the main site)

and vim mode still works

File: tooling/scripts/dl-monaco.ts [hunk 1]
```diff
@@ -46,14 +46,41 @@ async function downloadAndSaveFile([remoteUrl, diskPath = remoteUrl, appendToSta
...
 
 const files = [
   [
     '/esm/vs/editor/editor.api.d.ts',
-    '../../../monaco-editor.d.ts',
-    `// GENERATED WITH TS: ${TYPESCRIPT_VERSION}\n`,
+    '../../../packages/monaco/monaco-editor.d.ts',
+    (file) => {
+      // removes the global scope modifications since they need to be changed
+      const module = file.slice(837);
```
Comment (Hacksore) [inline tooling/scripts/dl-monaco.ts:58]: magic number 🥴

Comment (ggrandi) [reply tooling/scripts/dl-monaco.ts:58]:
```js
`/*!-----------------------------------------------------------
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Type definitions for monaco-editor
 * Released under the MIT license
*-----------------------------------------------------------*/
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

declare global {
    let MonacoEnvironment: Environment | undefined;

    interface Window {
        MonacoEnvironment?: Environment | undefined;
    }
}

interface Window {
    MonacoEnvironment?: Environment | undefined;
}
`.length
```

**Pull request merged**